<robot name="windmill">
    <gazebo>
        <plugin name="ros_control" filename="libgazebo_ros_control.so">
            <robotNamespace>/</robotNamespace>
            <robotSimType>gazebo_ros_control/DefaultRobotHWSim</robotSimType>
        </plugin>
    </gazebo>
    <link name="base_link">
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <geometry>
               <box size = "2 2 10"/>
            </geometry>
            <material name="green">
                <color rgba = "0 1 0 .5"/>
            </material>
        </visual>
        <inertial>
            <origin xyz="0 0 0"/>
            <mass value="400" />
            <inertia ixx="3466.67" ixy="0" ixz="0" iyy="3466.67" iyz="0" izz="266.67" />
        </inertial>
        <collision>
                    <geometry>
                        <box size="2 2 10" />
                    </geometry>
                    <origin xyz="0 0 0" rpy="0 0 0" />
        </collision>
    </link>
    <link name = "stick">
        <visual>
            <origin xyz="0 0 0" rpy = "0 0 0"/>
            <geometry>
                <cylinder radius="1" length = "6"/>
            </geometry>
        </visual>
       <inertial>
            <origin xyz="0 0 0"/>
            <mass value="3" />
            <inertia ixx="9.75" ixy="0" ixz="0" iyy="9.75" iyz="0" izz="1.5" />
        </inertial>
        <collision>
                    <geometry>
                        <cylinder radius="1" length = "6"/>
                    </geometry>
                    <origin xyz="0 0 0" rpy="0 0 0" />
        </collision>
    </link>

    <joint name="joint1" type="continuous">
        <origin xyz = "0 2 5" rpy="0 0 0"/>
        <parent link="base_link"/>
        <child link="stick"/>
        <axis xyz="0 1 0"/>
    </joint>
    <transmission name="left_front_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name ="joint1">
        <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="joint1_motor">
        <mechanicalReduction>1</mechanicalReduction>
        <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </actuator>
    </transmission>

</robot>